src/main/kotlin/com/atlassian/performance/tools/workspace/api/RootWorkspace.kt

```diff
@@ -19,20 +19,24 @@ class RootWorkspace(
      * The directory containing the root workspace.
      * Embedded within [parent].
      */
-    val directory: Path = parent
-        .resolve("jpt-workspace")
-        .toAbsolutePath()
-        .normalize()
-        .ensureDirectory()
+    val directory: Path by lazy {
+        parent
+            .resolve("jpt-workspace")
+            .toAbsolutePath()
+            .normalize()
+            .ensureDirectory()
+    }
 
     /**
      * The current task workspace unique to this root workspace instance.
      */
-    val currentTask: TaskWorkspace = isolateTask(
-        label = System.getenv("bamboo_buildResultKey")
-            ?: LocalDateTime.now()
-                .format(ISO_LOCAL_DATE_TIME)
-    )
+    val currentTask: TaskWorkspace by lazy {
```

Comment:
Once we have a lazy `Path`, here we have a lazy `TaskWorkspace`. Let's make it consistent, e.g. lazy `Path` fields.

Reply:
Which fields do you mean? The only one not lazy in this class is `private val parent` but creating it has no side effect so it can stay as it is

Reply:
Ah, so the pattern is: `lazy` around `ensureDirectory`?

Reply:
I mean, `RootWorkspace` represents its `directory` field. Same for `TaskWorkspace`.
But the root one ensures its existence, but the task one doesn't.